crossfire/clients/occurrences.py

```diff
@@ -164,3 +177,43 @@ def __call__(self):
             return GeoDataFrame(self.data)
 
         return self.data
+
+
+def _flatten_df(row, column_name):
+    column_data = row[column_name]
+    return Series(
+        {f"{column_name}_{key}": value for key, value in column_data.items()}
+    )
+
+
+def is_empty(data):
+    if HAS_PANDAS and isinstance(data, DataFrame):
+        return data.empty
+    return not data
+
+
+def flatten(data, nested_columns=None):
+    nested_columns = set(nested_columns or NESTED_COLUMNS)
+    if not nested_columns.issubset(NESTED_COLUMNS):
+        raise NestedColumnError(nested_columns)
+    if is_empty(data):
+        return data
+    if HAS_PANDAS and isinstance(data, DataFrame):
+        keys = set(data.columns) & nested_columns
+        for key in keys:
+            data = concat(
+                (
+                    data.drop(key, axis=1),
+                    data.apply(_flatten_df, args=(key,), axis=1),
+                ),
+                axis=1,
+            )
```

Comment:
Don't you think all this logic should be `_flatten_df`?

You can have the auxiliary function inside it, e.g.:

```python
def _flatten_df(data):
    def _flatten_col(col_name):
        # ...
    # code like data.apply(_flatten_col, col_name)
```